packages/client/hmi-client/src/api/api.ts

```diff
@@ -0,0 +1,7 @@
+import axios from 'axios';
+
+const API = axios.create({
+	baseURL: '/api'
+});
```

Comment:
Could we add the Keycloack Token as well?

Reply:
I tried to do this but I think it might require some very roundabout code.

We want to create a new axios instance when we create the Vue app instance which means getting `auth.token` then. But `auth.token` can change or expire which makes me think we'd actually want to get `auth.token` from `useAuthStore` each time we make a request.

That would involve wrapping/replacing axios's request functions... which seems like a much more complicated of just doing something like what we currently have in `uncloak()`. Am I missing something?

Reply:
Good questions!
@bigglesandginger do you have any insights?

Reply:
How did this work without the token?
I'm still confused as to why we can't just use the `uncloak` util, or at least use one or the other (I don't think we need both).

If using axios have you tried adding the
`headers: {'Authorization': 'Bearer '+token}` where `token` is retrieved from the store? I doubt it will dynamically evaluate it then but worth a try.

The other option (although probably not the best) is to use an interceptor and add it then.

Again I may be missing something key here though.

Reply:
I'd also love to know how this worked without the token 😆 I just know that it did. Happy to jump on a call to walk through it together if you'd be able to help me understand that.

I'm also not clear on why we need axios for our current usage, it does feel like overkill at the moment. @YohannParis mentioned that @dgauldie and @mwdchang both wanted to add it so maybe they can chime in.

I tried exactly what you suggested and got errors where the store hadn't been created yet. I figured it wasn't worth spending a bunch of time troubleshooting given that we're going to have the dynamic evaluation issue anyway.

If I understand what you're suggesting with the "interceptor" idea, I had the same thought. At that point seems like a lot of potentially confusing code to do what we're already doing with `uncloak`.

Reply:
I haven't looked at Ahmed PR so I am not up-to-date with the cloak/uncloak stuff.

Reply:
its just a wrapper around a `fetch` call that adds the token to the request nothing complicated there.
looks like he modified it slightly to allow for more than just GET requests

Reply:
Yeah I took a look and got the context, so basically if we want axios, `uncloak` should be like a helper wrapper  function that uses axios under the hood.

Reply:
@mwdchang do you see any benefit to using axios under the hood of `uncloak`? It's working fine now without.

Reply:
I think we have agreed to leverage axios, so I will close this thread.